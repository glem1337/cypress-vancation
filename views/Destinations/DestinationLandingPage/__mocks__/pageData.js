export default {
  id: 12345,
  coordinates: {
    latitude: 40.7306,
    longitude: -73.9866,
    searchRadius: 5,
  },
  mainInfo: {
    locationName: 'New York',
    stateName: 'NY',
    stateSlug: 'new-york',
  },
  topLocations: {},
  fanFacts: {},
  nearbyDestinations: {
    isVisible: false,
  },
};
