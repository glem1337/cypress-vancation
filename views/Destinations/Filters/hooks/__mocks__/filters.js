export default {
  uuid: null,
  seats: null,
  sleeps: null,
  priceStart: null,
  priceEnd: null,
  vehicles: null,
  delivery: null,
  glamper: null,
  rating: null,
  allowPets: null,
  allowSmoking: null,
  festivalApproved: null,
  allowUnlimitedMiles: null,
  insideHeight: null,
  standardAmenities: null,
  luxuryAmenities: null,
};
