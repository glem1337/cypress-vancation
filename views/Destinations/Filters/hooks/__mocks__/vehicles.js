export default [
  {
    id: 'e966978f-b6eb-4751-bdb7-e160ed9487cf',
    name: 'Unique Camper',
    iconUrl: 'https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/vehicletype/e966978f-b6eb-4751-bdb7-e160ed9487cf/icon/eeef67069b658ab4b051a22d3a128ab4.svg',
    actualCashValue: 40000,
    description: 'Skoolies, Ambulances and Other Unique Camper Conversions.',
    estimatedEarning: 1000,
  },
  {
    id: '76418300-edf1-475a-bd3c-34ba14723a0a',
    name: 'VW Bus',
    iconUrl: 'https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/vehicletype/76418300-edf1-475a-bd3c-34ba14723a0a/icon/ee7ec84b523ec774633950b3ca441ffe.svg',
    actualCashValue: 35000,
    description: 'Iconic and Classic. Westfalias, Vanagons, Eurovans & More Volkswagen Vans.',
    estimatedEarning: 900,
  },
  {
    id: 'fa58e13e-75d1-439f-a525-c337c4b48872',
    name: 'Vehicle Camper',
    iconUrl: 'https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/vehicletype/fa58e13e-75d1-439f-a525-c337c4b48872/icon/859293aa737ffc99491fcae32786030f.svg',
    actualCashValue: 35000,
    description: 'Truck Camper Rigs. Car, Minivan, Jeep and SUV conversions, typically with roof top tents.',
    estimatedEarning: 750,
  },
  {
    id: 'def880bc-0e47-4af0-a26f-315099640df3',
    name: 'Modern Van',
    iconUrl: 'https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/vehicletype/def880bc-0e47-4af0-a26f-315099640df3/icon/36d5c37d87f3c669cee5652ec00220c9.svg',
    actualCashValue: 65000,
    description: 'Sprinters, Transits, Promasters & More Fully Built Out Camper Vans.',
    estimatedEarning: 1400,
  },
];
