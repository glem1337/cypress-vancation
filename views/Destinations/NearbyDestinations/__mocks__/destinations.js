export default [
  {
    id: '32fb2587-2dfb-40b9-9c81-584822b0ac7f',
    landingName: 'New York',
    landingSlug: 'new-york',
    landingType: 'state_landing',
    latitude: 40.73061,
    longitude: -73.935242,
    searchRadius: 4,
    stateName: null,
    stateSlug: null,
    mainPhotoUrl: 'https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/statelanding/32fb2587-2dfb-40b9-9c81-584822b0ac7f/main_photo/61fd9bccd8fb04fe0dfb15688f474e6b.jpg',
    mainPhotoUrl360: 'https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/statelanding/32fb2587-2dfb-40b9-9c81-584822b0ac7f/main_photo/61fd9bccd8fb04fe0dfb15688f474e6b.jpg',
  },
];
