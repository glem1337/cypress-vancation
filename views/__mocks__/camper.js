export default {
  id: 'a18f7338-c2b3-4cee-be4d-579c0a726c1a',
  name: 'test',
  description: 'desc',
  status: 'draft',
  insurance: 'pending',
  place: 'Astoria, New York, New York 11102, United States',
  publicId: '111',
  estimatedEarning: 1200.0,
  vehicleTypeIconUrl: 'test.svg',
  vehicleTypeName: 'Modern Van',
  placeId: 'neighborhood.290416',
  longitude: -73.93030000000002,
  latitude: 40.772,
  glamper: true,
  owner: {
    id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
    account: {
      createdAt: '2021-07-09T13:39:26.587Z',
      drivingLicenceVerified: false,
      email: 'kateryna.sessa+owner@rubygarage.org',
      emailVerified: false,
      id: '12563773-6e8f-4861-8abd-82a78632c6ac',
      user: {
        avatarUrl: null,
        createdAt: '2021-07-09T13:39:26.590Z',
        firstName: 'Kat',
        id: '77015afb-71e3-4b94-a9cb-69a97dc6b358',
        lastName: 'Kat',
        updatedAt: '2021-07-09T13:39:26.590Z',
      },
    },
    type: 'owner',
    avatarUrl: null,
    averageRating: null,
    businessTitle: null,
    campersCount: 25,
    createdAt: 'Member since July 2021',
    description: 'description',
    email: null,
    idVerified: false,
  },
  specificationDetail: {
    drivetrain: '4x4',
    fuelType: 'electric',
    insideHeight: 'low_top',
    length: '67.89',
    modelNaming: 'Clair Weimann',
    name: 'Katelyn Lowe',
    seats: 1,
    sleeps: 1,
    stateRegistred: 'Arkansas',
    transmission: 'automatice',
    year: 1585,
    mileage: 'Under 50,000',
    whoBuiltCamper: 'Vasya',
  },
  pricingInfo: {
    id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
  },
  camperRule: {
    id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
    bookingApprovalPolicy: 'instant_book',
    cancellationPolicy: 'easy_going',
  },
  deliveryInformation: {
    id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
    pickup: true,
  },
  camperPhotos: [{
    id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
    photoUrl1100: 'photoUrl1100',
  }],
  insuranceInfo: {
    id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
  },
  tripFee: {
    id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
    cleaning: 1,
    dumpingFee: 2,
    flue: 3,
    lateDropOff: 4,
    customFees: [
      {
        id: 'id',
        name: 'name',
        price: 5,
        frequency: 'per_day',
      },
    ],
    tripFeeMileage: {
      limit: true,
      available: 6,
      overage: 7,
    },
    tripFeeGenerator: {
      limit: true,
      available: 8,
      overage: 9,
    },
  },
  blockedPeriods: [],
  pricingPeriods: [],
  externalCalendars: [],
  camperCalendar: {
    id: 'e1db81b1-8ad5-4e88-9c99-f5c9020148eb',
    externalCalendars: [],
    blockedPeriods: [
      {
        id: '56ceb535-322f-42c6-8dcd-a88ad59000f3',
        startDate: '2021-08-05',
        endDate: '2021-08-06',
      },
      {
        id: '17a32ec0-a486-4247-910c-6c0c57ba6ec3',
        startDate: '2021-08-25',
        endDate: '2021-08-25',
      },
      {
        id: '2c7bbaef-6b00-494f-b005-6a762e90c785',
        startDate: '2021-07-28',
        endDate: '2021-07-28',
      },
    ],
    pricingPeriods: [
      {
        id: '8aa92a51-ef00-4dc6-8b84-2820869b9278',
        startDate: '2021-07-28',
        endDate: '2021-07-28',
        price: '33.0',
      },
      {
        id: '248ba38e-e18a-467c-996b-36aa4bd73e85',
        startDate: '2021-07-29',
        endDate: '2021-07-30',
        price: '320.0',
      },
    ],
    customDiscountPeriods: [
      {
        id: 'ffb3f0cc-4513-4fdc-81d0-03587232d9ee',
        startDate: '2021-07-27',
        endDate: '2021-07-28',
        weeklyDiscountPercent: 15,
        monthlyDiscountPercent: 23,
      },
      {
        id: 'a070339d-7058-4769-8e16-814084c6bfe3',
        startDate: '2021-07-28',
        endDate: '2021-07-28',
        weeklyDiscountPercent: 15,
        monthlyDiscountPercent: 23,
      },
      {
        id: 'aba16cba-5873-4273-8172-816b276a2db7',
        startDate: '2021-07-28',
        endDate: '2021-07-28',
        weeklyDiscountPercent: 44,
        monthlyDiscountPercent: 44,
      },
      {
        id: '586016b9-d351-4ca1-ac8a-373e7e3b8832',
        startDate: '2021-07-29',
        endDate: '2021-07-29',
        weeklyDiscountPercent: 15,
        monthlyDiscountPercent: 33,
      },
    ],
  },
  camperAddition: {
    amenityHealthSafetyItems: [
      {
        healthSafety: {
          id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
          iconUrl: 'path/to/icon/url',
          name: 'name',
        },
      },
    ],
    restrictionRule: {
      allowPets: false,
      smoking: false,
      festivalApproved: false,
    },
    customRestrictionRules: [
      {
        id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
        active: true,
        name: 'name',
      },
    ],
    travelRestriction: {
      mexico: false,
      canada: true,
      burningMan: false,
    },
    customTravelRestrictions: [
      {
        id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
        active: true,
        name: 'name',
      },
    ],
    restrictionRoad: {
      fourWheelRoad: false,
      offRoad: false,
      snowAndIceRoad: false,
      dirtryRoad: true,
    },
    customRestrictionRoads: [
      {
        id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
        active: true,
        name: 'name',
      },
    ],
    camperQuestions: [
      {
        id: '74a9c587-7b78-4aa5-ba93-ba3d310be127',
        text: 'test',
        required: true,
      },
    ],
    customTravelAccessories: [
      {
        active: true,
        description: '',
        id: 'dd477886-ec22-4da9-ba4c-08627f35fb1a',
        maxAmount: '2.0',
        name: 'Bike Rack',
        price: '1.0',
        priceUnit: 'each',
      },
    ],
  },
};
