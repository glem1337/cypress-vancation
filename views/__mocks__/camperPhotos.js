export default [
  {
    id: 1,
    mainPhoto: true,
    photo: 'http://.../uploads/store/camperphoto',
    camper: {
      id: '0aebbaa8-5006-483b-962c-82fe01ed581f',
      status: 'draft',
    },
  },
  {
    id: 2,
    mainPhoto: false,
    photo: 'http://.../uploads/store/camperphoto',
    camper: {
      id: '0aebbaa8-5006-483b-962c-82fe01ed581f',
      status: 'draft',
    },
  },
];
