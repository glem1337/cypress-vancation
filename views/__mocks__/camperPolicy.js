export default {
  booking_approval_policy: 'instant_book',
  cancellation_policy: 'easy_going',
  camper_id: 'f7445371-9f19-4c52-b0a4-711785cb55f1',
};
