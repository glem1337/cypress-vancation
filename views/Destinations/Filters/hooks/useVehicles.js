import { useState, useEffect, useRef } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import * as R from 'ramda';
import { Checkbox } from 'antd';
import { debounce } from 'lodash';

import { FILTERS } from 'constants/searchDestinations';
import { searchDestinationFiltersSelector } from 'state/concepts/search-destinations/selectors';
import { setFilterValue } from 'state/concepts/search-destinations/actions';
import { vehicleTypeSelector } from 'state/concepts/camper/selectors';

function useContainer(props) {
  const dispatch = useDispatch();

  const persistedFilters = useSelector(searchDestinationFiltersSelector);

  const vehicleTypes = useSelector(vehicleTypeSelector);

  const [filters, setFilters] = useState(persistedFilters.vehicles || []);

  /**
   * Persisted handlers.
   */
  const handlers = useRef({
    /**
     * Immediately Apply filters.
     */
    applyFiltersImmediately: debounce((value) => {
      dispatch(setFilterValue({
        name: 'vehicles',
        value,
      }));
    }, FILTERS.DEBOUNCE_TIME),
  });

  /**
   * Clear filters.
   */
  const clearFilters = () => {
    dispatch(setFilterValue({
      name: 'vehicles',
      value: null,
    }));

    props.toggleOpenedState(null)();
  };

  /**
   * Apply filters.
   */
  const applyFilters = () => {
    dispatch(setFilterValue({
      name: 'vehicles',
      value: filters,
    }));

    props.toggleOpenedState(null)();

    return filters;
  };

  /**
   * On change handler.
   */
  const onChangeHandler = (id) => () => {
    let array = [...filters];

    const index = array.indexOf(id);

    if (index !== -1) {
      array = R.remove(index, 1, array);
    }

    if (index === -1) {
      array = [...array, id];
    }

    setFilters(array);

    if (props.immediatelyApply) {
      handlers.current.applyFiltersImmediately(array);
    }
  };

  /**
   * Detect is checked.
   */
  const isChecked = (id) => R.includes(id, filters);

  /**
   * Render vehicle widget.
   */
  const renderVehiclesWidget = () => (
    <>
      {R.defaultTo([], vehicleTypes).map(item => (
        <Checkbox
          key={item.id}
          className="main-checkbox-card"
          onChange={onChangeHandler(item.name)}
          checked={isChecked(item.name)}
        >
          <i className="icon icon-activate-f main-checkbox-card__icon" />
          <div className="d-flex flex-column flex-1">
            <div className="d-flex align-items-center mb-16">
              <img src="/images/listing/Modern-Van.svg" alt="" />
              <span className="text-subheader font-400 ml-8">{item.name}</span>
            </div>
            <div className="mr-40">{item.description}</div>
          </div>
        </Checkbox>
      ))}
    </>
  );

  /**
   * Check filter updates
   */
  const checkFilterUpdates = () => {
    if (!persistedFilters.uuid) {
      return false;
    }

    setFilters(persistedFilters.vehicles || []);

    return true;
  };

  /**
   * Updating
   */
  useEffect(() => {
    checkFilterUpdates();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [persistedFilters.uuid]);

  /**
   * Updating
   */
  useEffect(() => {
    checkFilterUpdates();
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [props.openedFilterName]);

  return {
    vehicleTypes,
    clearFilters,
    applyFilters,
    isChecked,
    onChangeHandler,
    badge: persistedFilters.vehicles?.length,
    renderVehiclesWidget,
    setFilters,
    filters,
    checkFilterUpdates,
  };
}

export default useContainer;
