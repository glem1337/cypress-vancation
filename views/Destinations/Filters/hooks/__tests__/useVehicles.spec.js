import { act, renderHook } from '@testing-library/react-hooks';

import { dispatch } from '__mocks__/react-redux';
import { setFilterValue } from 'state/concepts/search-destinations/actions';
import { searchDestinationFiltersSelector } from 'state/concepts/search-destinations/selectors';

import useVehicles from '../useVehicles';
import mockedFilter from '../__mocks__/filters';
import mockedVehicles from '../__mocks__/vehicles';

jest.mock('state/concepts/search-destinations/selectors', () => ({
  searchDestinationFiltersSelector: jest.fn(() => mockedFilter),
}));

jest.mock('state/concepts/campervan-rental/selectors', () => ({
  campersTotalSelector: jest.fn(() => 22),
}));

jest.mock('state/concepts/camper/selectors', () => ({
  vehicleTypeSelector: jest.fn(() => mockedVehicles),
}));

jest.mock('state/data/selectors', () => ({
  loadingSelector: jest.fn(() => true),
}));

jest.mock('lodash/debounce', () => fn => fn);

describe('useVehicles hook', () => {
  let result = null;

  const props = {
    toggleOpenedState: jest.fn(() => jest.fn()),
  };

  beforeEach(() => {
    searchDestinationFiltersSelector.mockReturnValue(mockedFilter);

    ({ result } = renderHook(() => useVehicles(props)));

    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot();
  });

  it('checks `clearFilters` method', () => {
    act(() => {
      result.current.clearFilters();
    });

    expect(dispatch).toHaveBeenCalledWith(setFilterValue({
      name: 'vehicles',
      value: null,
    }));

    expect(props.toggleOpenedState).toHaveBeenCalled();
  });

  it('checks `applyFilters` method', () => {
    let filters = null;

    act(() => {
     filters = result.current.applyFilters();
    });

    expect(dispatch).toHaveBeenCalledWith(setFilterValue({
      name: 'vehicles',
      value: filters,
    }));

    expect(props.toggleOpenedState).toHaveBeenCalled();
  });

  describe('checks `onChangeHandler` method', () => {
    it('should add array item', () => {
      act(() => {
        result.current.onChangeHandler('test')();
      });

      expect(result.current.filters).toEqual(['test']);
    });

    it('should remove array item', () => {
      act(() => {
        result.current.setFilters(['test']);
      });

      act(() => {
        result.current.onChangeHandler('test')();
      });

      expect(result.current.filters).toEqual([]);
    });

    it('with immediate update', () => {
      ({ result } = renderHook(() => useVehicles({
        ...props,
        immediatelyApply: true,
      })));

      act(() => {
        result.current.onChangeHandler('test')();
      });

      expect(dispatch).toHaveBeenCalledWith(setFilterValue({
        name: 'vehicles',
        value: ['test'],
      }));
    });
  });

  describe('checks `isChecked` method', () => {
    it('should return false', () => {
      let res = null;

      act(() => {
        res = result.current.isChecked('test');
      });

      expect(res).toBe(false);
    });

    it('should return true', () => {
      act(() => {
        result.current.setFilters(['test']);
      });

      let res = null;

      act(() => {
        res = result.current.isChecked('test');
      });

      expect(res).toBe(true);
    });
  });

  it('checks `renderVehiclesWidget` method', () => {
    let res = null;

    act(() => {
      res = result.current.renderVehiclesWidget();
    });

    expect(res).toMatchSnapshot();
  });

  describe('checks `checkFilterUpdates` method', () => {
    it('should return false', () => {
      let res = null;

      act(() => {
        res = result.current.checkFilterUpdates();
      });

      expect(res).toBe(false);
    });

    it('should return true', () => {
      searchDestinationFiltersSelector.mockReturnValue({
        ...mockedFilter,
        uuid: '213218n-vdjiks',
      });

      ({ result } = renderHook(() => useVehicles(props)));

      let res = null;

      act(() => {
        res = result.current.checkFilterUpdates();
      });

      expect(res).toBe(true);
    });
  });
});
