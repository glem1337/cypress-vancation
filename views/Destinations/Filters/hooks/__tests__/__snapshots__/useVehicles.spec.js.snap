// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useVehicles hook checks \`renderVehiclesWidget\` method 1`] = `
<React.Fragment>
  <Checkbox
    checked={false}
    className="main-checkbox-card"
    onChange={[Function]}
  >
    <i
      className="icon icon-activate-f main-checkbox-card__icon"
    />
    <div
      className="d-flex flex-column flex-1"
    >
      <div
        className="d-flex align-items-center mb-16"
      >
        <img
          alt=""
          src="/images/listing/Modern-Van.svg"
        />
        <span
          className="text-subheader font-400 ml-8"
        >
          Unique Camper
        </span>
      </div>
      <div
        className="mr-40"
      >
        Skoolies, Ambulances and Other Unique Camper Conversions.
      </div>
    </div>
  </Checkbox>
  <Checkbox
    checked={false}
    className="main-checkbox-card"
    onChange={[Function]}
  >
    <i
      className="icon icon-activate-f main-checkbox-card__icon"
    />
    <div
      className="d-flex flex-column flex-1"
    >
      <div
        className="d-flex align-items-center mb-16"
      >
        <img
          alt=""
          src="/images/listing/Modern-Van.svg"
        />
        <span
          className="text-subheader font-400 ml-8"
        >
          VW Bus
        </span>
      </div>
      <div
        className="mr-40"
      >
        Iconic and Classic. Westfalias, Vanagons, Eurovans & More Volkswagen Vans.
      </div>
    </div>
  </Checkbox>
  <Checkbox
    checked={false}
    className="main-checkbox-card"
    onChange={[Function]}
  >
    <i
      className="icon icon-activate-f main-checkbox-card__icon"
    />
    <div
      className="d-flex flex-column flex-1"
    >
      <div
        className="d-flex align-items-center mb-16"
      >
        <img
          alt=""
          src="/images/listing/Modern-Van.svg"
        />
        <span
          className="text-subheader font-400 ml-8"
        >
          Vehicle Camper
        </span>
      </div>
      <div
        className="mr-40"
      >
        Truck Camper Rigs. Car, Minivan, Jeep and SUV conversions, typically with roof top tents.
      </div>
    </div>
  </Checkbox>
  <Checkbox
    checked={false}
    className="main-checkbox-card"
    onChange={[Function]}
  >
    <i
      className="icon icon-activate-f main-checkbox-card__icon"
    />
    <div
      className="d-flex flex-column flex-1"
    >
      <div
        className="d-flex align-items-center mb-16"
      >
        <img
          alt=""
          src="/images/listing/Modern-Van.svg"
        />
        <span
          className="text-subheader font-400 ml-8"
        >
          Modern Van
        </span>
      </div>
      <div
        className="mr-40"
      >
        Sprinters, Transits, Promasters & More Fully Built Out Camper Vans.
      </div>
    </div>
  </Checkbox>
</React.Fragment>
`;

exports[`useVehicles hook matches snapshot 1`] = `
Object {
  "applyFilters": [Function],
  "badge": undefined,
  "checkFilterUpdates": [Function],
  "clearFilters": [Function],
  "filters": Array [],
  "isChecked": [Function],
  "onChangeHandler": [Function],
  "renderVehiclesWidget": [Function],
  "setFilters": [Function],
  "vehicleTypes": Array [
    Object {
      "actualCashValue": 40000,
      "description": "Skoolies, Ambulances and Other Unique Camper Conversions.",
      "estimatedEarning": 1000,
      "iconUrl": "https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/vehicletype/e966978f-b6eb-4751-bdb7-e160ed9487cf/icon/eeef67069b658ab4b051a22d3a128ab4.svg",
      "id": "e966978f-b6eb-4751-bdb7-e160ed9487cf",
      "name": "Unique Camper",
    },
    Object {
      "actualCashValue": 35000,
      "description": "Iconic and Classic. Westfalias, Vanagons, Eurovans & More Volkswagen Vans.",
      "estimatedEarning": 900,
      "iconUrl": "https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/vehicletype/76418300-edf1-475a-bd3c-34ba14723a0a/icon/ee7ec84b523ec774633950b3ca441ffe.svg",
      "id": "76418300-edf1-475a-bd3c-34ba14723a0a",
      "name": "VW Bus",
    },
    Object {
      "actualCashValue": 35000,
      "description": "Truck Camper Rigs. Car, Minivan, Jeep and SUV conversions, typically with roof top tents.",
      "estimatedEarning": 750,
      "iconUrl": "https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/vehicletype/fa58e13e-75d1-439f-a525-c337c4b48872/icon/859293aa737ffc99491fcae32786030f.svg",
      "id": "fa58e13e-75d1-439f-a525-c337c4b48872",
      "name": "Vehicle Camper",
    },
    Object {
      "actualCashValue": 65000,
      "description": "Sprinters, Transits, Promasters & More Fully Built Out Camper Vans.",
      "estimatedEarning": 1400,
      "iconUrl": "https://vancation-api-staging-assets.s3.us-west-1.amazonaws.com/store/vehicletype/def880bc-0e47-4af0-a26f-315099640df3/icon/36d5c37d87f3c669cee5652ec00220c9.svg",
      "id": "def880bc-0e47-4af0-a26f-315099640df3",
      "name": "Modern Van",
    },
  ],
}
`;
